### PR TITLE
feat: get lambda-url to directly access fn over http

### PR DESCRIPTION
**Pull request by olizilla**
Gets us a generated url to directly invoke the linkdex-api lambda, to bypass the api gateway. This lets us make use of the full 15 min lambda max duration. When we go via the aws api gateway it applies a 30s max timeout on all requests.

see: https://docs.aws.amazon.com/lambda/latest/dg/lambda-urls.html

### CODE REVIEW

File: stacks/LinkdexStack.ts [hunk 1]
```diff
@@ -10,23 +10,25 @@ export function LinkdexStack({ app, stack }: StackContext) {
...
 
   const customDomain = getCustomDomain(app.stage, process.env.HOSTED_ZONE)
+
+  const lindexKeyFn = new Function(stack, 'LindexKeyFn', {
+    handler: 'functions/linkdex.handler',
+    permissions: [bucket],
+    environment: { BUCKET_NAME: bucket.bucketName },
+    architecture: 'arm_64', // cheaper, so why not?
+    timeout: '15 minutes',  // default is 10s. Api gateway has 30s limit. Lambda max 15 mins.
+    url: true
```
Comment (olizilla) [inline stacks/LinkdexStack.ts:20]: This `url: true` is the magic config option that gets aws to assign a direct lambda url for the function.

Comment (olizilla) [reply stacks/LinkdexStack.ts:20]: but you can't set it via api gateway config, so we create our own function construct here and pass it to the api.

Comment (olizilla) [reply stacks/LinkdexStack.ts:20]: We could drop the api gateway altogether but it's kinda handy for the nice per PR custom domains.

Comment (vasco-santos) [reply stacks/LinkdexStack.ts:20]: who wow, interesting that using the "correct way of doing aws" with gateway in front leads to such a smaller timeout

**Review by vasco-santos — APPROVED** [review]
LGTM

**Pull request merged**